### PR TITLE
field: reinforcement of tests with edge cases and proptests

### PR DESCRIPTION
**Pull request by tcoratger**
After https://github.com/Plonky3/Plonky3/pull/1429, I thought it would be nice to add some proptests for the fields as well as some missing edge cases.

### CODE REVIEW

File: field-testing/src/lib.rs [hunk 1]
```diff
@@ -324,6 +331,9 @@ where
 {
     assert_eq!(None, F::ZERO.try_inverse());
     assert_eq!(Some(F::ONE), F::ONE.try_inverse());
+    assert_eq!(F::NEG_ONE.inverse(), F::NEG_ONE, "-1 is its own inverse");
+    let two_inv = F::TWO.try_inverse().unwrap();
```
[the remark this answers is not in the document]
Comment (tcoratger) [reply field-testing/src/lib.rs]: Fixed

File: field-testing/src/lib.rs [hunk 2]
```diff
@@ -222,6 +235,17 @@ where
...
 
+    // Edge case tests with special values
+    for &a in &[R::ZERO, R::ONE, R::TWO, R::NEG_ONE] {
+        for &b in &[R::ZERO, R::ONE, R::TWO, R::NEG_ONE] {
+            assert_eq!(a + b, b + a, "commutativity with special values");
+            assert_eq!(a * b, b * a, "commutativity with special values");
+            assert_eq!(a * a, a.square(), "square with special value");
+            assert_eq!(a * a * a, a.cube(), "cube with special value");
+            assert_eq!(a.halve().double(), a, "halve/double with special value");
+        }
```
[the remark this answers is not in the document]
Comment (tcoratger) [reply field-testing/src/lib.rs]: Fixed

**Comment by tcoratger** [comment]
@Nashtare @adr1anh Is there anything missing here to merge? 

Not a big deal but I feel this is nice to have in order to secure our future PRs against potential edge case bugs :)

File: field-testing/src/lib.rs [hunk 3]
```diff
@@ -881,6 +893,81 @@ where
     assert_eq!(combined_u64s, expected_combined_u64s);
 }
 
+/// Test ring axioms with 256 random (x, y, z) triplets via proptest.
+///
+/// Tests commutativity, associativity, distributivity, negation,
+/// subtraction identities, square/cube, and double/halve.
+pub fn test_ring_axioms_proptest<R>()
```
Comment (SyxtonPrime) [inline field-testing/src/lib.rs:901]: Maybe also throw in a check that `-x = -1 * x` and `0 = x * 0`.

File: field-testing/src/lib.rs [hunk 4]
```diff
@@ -881,6 +893,81 @@ where
...
+    assert_eq!(F::NEG_ONE.inverse(), F::NEG_ONE, "-1 is its own inverse");
+    assert_eq!(
+        F::GENERATOR.inverse() * F::GENERATOR,
+        F::ONE,
+        "generator inverse roundtrip"
+    );
+
+    // Proptest: 256 random pairs
+    let config = ProptestConfig::with_cases(256);
+    proptest!(config, |(x in arb_field::<F>(), y in arb_field::<F>())| {
```
Comment (SyxtonPrime) [inline field-testing/src/lib.rs]: Maybe pick `3` elements and just start by assuming all are non zero so that we don't need too many if statements.

Then we can throw in the checks `x / (y * z) = (x / y ) / z` and `(x * y) / z = x * (y / z)`

File: field-testing/src/lib.rs (hunk 3 above)
Comment (tcoratger) [reply field-testing/src/lib.rs:901]: Fixed

File: field-testing/src/lib.rs (hunk 4 above)
Comment (tcoratger) [reply field-testing/src/lib.rs]: Fixed
